test/widgets/subscription_list_test.dart

```diff
@@ -89,18 +89,49 @@ void main() {
     check(isUnpinnedHeaderInTree()).isTrue();
   });
 
-  testWidgets('subscription sort', (tester) async {
-    await setupStreamListPage(tester, subscriptions: [
-      eg.subscription(eg.stream(streamId: 1, name: 'd'), pinToTop: true),
-      eg.subscription(eg.stream(streamId: 2, name: 'c'), pinToTop: false),
-      eg.subscription(eg.stream(streamId: 3, name: 'b'), pinToTop: true),
-      eg.subscription(eg.stream(streamId: 4, name: 'a'), pinToTop: false),
-    ]);
-    check(isPinnedHeaderInTree()).isTrue();
-    check(isUnpinnedHeaderInTree()).isTrue();
-
-    final streamListItems = tester.widgetList<SubscriptionItem>(find.byType(SubscriptionItem)).toList();
-    check(streamListItems.map((e) => e.subscription.streamId)).deepEquals([3, 1, 4, 2]);
+  group('subscription sorting', () {
+    streamListIds(WidgetTester tester) => tester
+      .widgetList<SubscriptionItem>(find.byType(SubscriptionItem))
+      .map((e) => e.subscription.streamId);
+
+    testWidgets('pinned are shown on the top', (tester) async {
+      await setupStreamListPage(tester, subscriptions: [
+        eg.subscription(eg.stream(streamId: 1, name: 'a'), pinToTop: false),
+        eg.subscription(eg.stream(streamId: 2, name: 'b'), pinToTop: true),
+        eg.subscription(eg.stream(streamId: 3, name: 'c'), pinToTop: false),
+      ]);
+      check(streamListIds(tester)).deepEquals([2, 1, 3]);
```

Comment:
This name `streamListIds`, or "stream list IDs", sounds like it'd describe things that were IDs of "stream lists", or "list IDs" that were related to streams.

But in fact these are IDs of streams, namely of the streams shown in the list.

In the old code `streamListItems` worked as a name because each element was an item (a `SubscriptionItem`, in fact) in the stream list (the `StreamListPage`). So, a "stream list item".

For this code, one name that would work is `listedStreamIds` as I suggested at https://github.com/zulip/zulip-flutter/pull/581#discussion_r1540332942. (Each element of this return value is a stream ID, specifically the ID of a stream that's listed in this stream list.) So let's use that.